src/escape.rs

```diff
@@ -0,0 +1,100 @@
+//! Escape characters that may have special meaning in a shell, including spaces.
+//! This is a modified version of the [`shell-escape::unix`] module of [`shell-escape`] crate.
+//! 
+//! [`shell-escape`]: https://crates.io/crates/shell-escape
+//! [`shell-escape::unix`]: https://docs.rs/shell-escape/latest/src/shell_escape/lib.rs.html#101
+
+use std::{
+    borrow::Cow, 
+    ffi::{OsStr, OsString},
+    os::unix::ffi::OsStrExt,
+    os::unix::ffi::OsStringExt,
+};
+
+
+fn whitelisted(byte: u8) -> bool {
+    match byte {
+        b'a'..=b'z' | b'A'..=b'Z' | b'0'..=b'9' | b'-' | b'_' | b'=' | b'/' | b',' | b'.' | b'+' => true,
+        _ => false,
+    }
+}
+
+/// Escape characters that may have special meaning in a shell, including spaces.
+/// 
+/// **Note**: This function is an adaptation of [`shell-escape::unix::escape`].
+/// This function exists only for type compatibility and the implementation is
+/// almost exactly the same as [`shell-escape::unix::escape`].
+/// 
+/// [`shell-escape::unix::escape`]: https://docs.rs/shell-escape/latest/src/shell_escape/lib.rs.html#101
+/// 
+pub fn escape(s: &OsStr) -> Cow<'_, OsStr> {
+    let s = s.as_bytes();
+    let all_whitelisted = s.iter().all(|x| whitelisted(*x));
+
+    if !s.is_empty() && all_whitelisted {
+        return OsString::from_vec(s.to_vec()).into();
+    }
+
+    let mut escaped = Vec::with_capacity(s.len() + 2);
+    escaped.push(b'\'');
+
+    for &b in s {
+        match b {
+            b'\'' | b'!' => {
+                escaped.push(b'\'');
+                escaped.push(b'\\');
+                escaped.push(b);
+                escaped.push(b'\'');
+            }
+            _ => escaped.push(b),
+        }
+    }
+    escaped.push(b'\'');
+    OsString::from_vec(escaped).into()
+}
+
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn test_escape_case(input: &str, expected: &str) {
```

Comment:
It would be great if we can also test some non-utf8 characters, but I don't consider it necessary for this PR to be merged.